design_documents/transactional_testing/transactional_testing.md

```diff
@@ -0,0 +1,75 @@
+# Transactional testing proposal
+
+## Context
+
+Some flows are not currently intuitively testable via standard testing methods, 
+like accounts logic (__validate__, __execute__ etc.), transaction rejections,
+fees, contract upgrade flows, and other e2e testing procedures.
+
+## Goal
+
+In order to facilitate that, we should consider adding a way to test
+the contracts in conditions as close to real networks conditions as possible.
+
+Also, the goal is to make the deployments testable and validation 
+of the deployments possible to the user. 
+
+The tests should be also runnable in a variety of environments, depending on the
+use case (namely katana, devnet, or testnet). This would increase user's confidence
+in the release process.
+
+## Proposed solution
```

Comment:
Would cheatcodes be possible with this solution? I'm assuming not.

Reply:
It could be possible if node implements them but those shouldn't be our focus atm

Reply:
Some functions like that are supported (in devnet for example)

Reply:
[add docs links for specific katana/devnet functions](https://github.com/foundry-rs/starknet-foundry/pull/1156/commits/38e44121eecb5edb6364ee410e87db89d1708341)
see this commit for docs links (supported extra functionalities)